Source/Core/Core/Core.cpp

```diff
@@ -671,6 +671,22 @@ void SaveScreenShot()
 		SetState(CORE_RUN);
 }
 
+void SaveScreenShot(std::string name)
+{
+	const bool bPaused = (GetState() == CORE_PAUSE);
+
+	SetState(CORE_PAUSE);
+
+	const std::string& gameId = SConfig::GetInstance().GetUniqueID();
+    std::string path = File::GetUserPath(D_SCREENSHOTS_IDX) + gameId + DIR_SEP_CHR;
+
+	name = StringFromFormat("%s%s.png", path.c_str(), name.c_str());
```

Comment:
This should be its own string instead of modifying function input directly (the function parameter can subsequently be made a const reference).